### PR TITLE
Fix: raise error when cooks outliers should have been refitted but were not

### LINKED ISSUE #46: Wald-test error if no Cook distance refit is performed
First,  I would like to thank you for this amazing tool!

The problem I am facing result of running the _stat_res.run_wald_test()_ function
I have followed the step-by-step tutorial with my own data. I skipped the Cook distance-based refit.
As a result, my original DeseqDataSet object DO NOT have the "_replaced_" attribute.

This cause an error in DeseqStats.py/run_wald_test (line 303) , where it checks if it needs to correct the p-values based on Cook refiting. 

My guess is that the _replaced_ attribute should be initialized (?maybe set to pd.Series[0]?) when calling the DeseqDataSet class object.

This is solved if I force dds.replaced = pd.Series[0] and/or if I run the _dds.refit()_ function

Let me know if you require further info to reproduce this issue

### PR DESCRIPTION
**Pull request by BorisMuzellec**
This PR addresses issue #46 by raising an `AssertionError` if one tries to initialize a `DeseqStats` object  from a `DeseqDataSet` with `refit_cooks=True` but whose outliers were not refit. 

This is detected by checking that the `DeseqDataSet` has a `replaced` attribute. 

A corresponding pytest was also added.

Closes #46.

### CODE REVIEW

File: pydeseq2/DeseqStats.py [hunk 1]
```diff
@@ -191,6 +191,15 @@ def __init__(
...
         self.joblib_verbosity = joblib_verbosity
 
+        # If the `refit_cooks` attribute of the dds object is True, check that outliers
+        # were actually refitted.
+        if dds.refit_cooks:
+            assert hasattr(dds, "replaced"), (
+                "dds has 'refit_cooks' set to True but Cooks outliers have not been "
+                "refitted. Please run 'dds.refit()' first or set 'dds.refit_cooks' "
+                "to False."
+            )
```
Comment (mandreux-owkin) [inline pydeseq2/DeseqStats.py:201]: Instead of an `AssertionError`, what about raising an `AttributeError`? Otherwise LGTM

Comment (BorisMuzellec) [reply pydeseq2/DeseqStats.py]: Good idea, an `AttributeError` might be more appropriate indeed

File: tests/test_edge_cases.py [hunk 2]
```diff
@@ -150,6 +150,40 @@ def test_indexes():
         )
 
 
+def test_cooks_not_refitted():
+    """Test that an AssertionError is thrown when a `DeseqStats` object is initialized
```
Comment (maikia) [inline tests/test_edge_cases.py:154]:
```suggestion
    """Test that an AttributeError is thrown when a `DeseqStats` object is initialized
```

**Review by maikia — APPROVED** [review]
LGTM
if in the description of the PR you use: closes #issue_number the issue will automatically be closed when the PR is merged ( there are other possible keywords: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue )

**Comment by BorisMuzellec** [comment]
> if in the description of the PR you use: closes #issue_number the issue will automatically be closed when the PR is merged ( there are other possible keywords: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue )

Thanks, I didn't know we could do that :)

**Pull request merged**